### PR TITLE
Test algebraic independence

### PR DESCRIPTION
**Pull request by wdecker**

### CODE REVIEW

File: test/Rings/mpoly_affine_algebras.jl [hunk 1]
```diff
@@ -129,3 +129,9 @@ end
     @test V[i] == rels[i](V2...)
   end
 end
+
+@testset "are_algebraically_independent" begin
+  R, (x, y) = polynomial_ring(QQ, ["x", "y"])
+  V = [x, y]
+  @test is_zero(are_algebraically_independent(V)[2])
```
Comment (lgoettgens) [inline test/Rings/mpoly_affine_algebras.jl]:
```suggestion
  fl, I = are_algebraically_independent(V)
  @test fl
  @test is_zero(I)
```
Maybe test both return values

File: src/Rings/mpoly-affine-algebras.jl [hunk 2]
```diff
@@ -1056,6 +1056,54 @@ function _subalgebra_membership_homogeneous(f::PolyRingElemT, v::Vector{PolyRing
...
+  @req coefficient_ring(R) isa Field "The coefficient ring must be a field"
+  @req all(x -> parent(x) === R, V) "The elements must have the same parent"
+  S, _ = polynomial_ring(coefficient_ring(R), length(V), "t")
+  phi = hom(S, R, V)
+  I = kernel(phi)
+  bo = true
+  if !iszero(I)
+    bo = false
+  end
+  return bo, kernel(phi)
```
Comment (lgoettgens) [inline src/Rings/mpoly-affine-algebras.jl]:
```suggestion
  return iszero(I), I
```
simplication

File: src/Rings/mpoly-affine-algebras.jl [hunk 3]
```diff
@@ -1056,6 +1056,54 @@ function _subalgebra_membership_homogeneous(f::PolyRingElemT, v::Vector{PolyRing
...
+(false, ideal(t1*t2))
+
+```
+"""
+function are_algebraically_independent(V::Vector{T}) where T <: Union{MPolyRingElem, MPolyQuoRingElem}
+  @req !isempty(V) "Input vector must not be empty"
+  R = parent(V[1])
+  @req coefficient_ring(R) isa Field "The coefficient ring must be a field"
+  @req all(x -> parent(x) === R, V) "The elements must have the same parent"
+  S, _ = polynomial_ring(coefficient_ring(R), length(V), "t")
```
Comment (lgoettgens) [inline src/Rings/mpoly-affine-algebras.jl]:
```suggestion
  S, _ = polynomial_ring(coefficient_ring(R), length(V), "t"; cached = false)
```
poly rings in library code should not be cached (cf. https://github.com/oscar-system/Oscar.jl/issues/2455)

File: test/Rings/mpoly_affine_algebras.jl [hunk 4]
```diff
@@ -129,3 +129,11 @@ end
     @test V[i] == rels[i](V2...)
   end
 end
+
+@testset "are_algebraically_independent" begin
+  R, (x, y) = polynomial_ring(QQ, ["x", "y"])
+  V = [x, y]
+  fl, I = are_algebraically_independent(V)
+  @test fl
+  @test is_zero(I)
```
Comment (afkafkafk13) [inline test/Rings/mpoly_affine_algebras.jl:138]: Please also do an algebraically dependent example in the test.

**Review by afkafkafk13 — APPROVED** [review]

**Comment by wdecker** [comment]
The non-examples are in the doc tests.Von meinem iPhone gesendetAm 15.09.2023 um 23:16 schrieb Anne Frühbis-Krüger ***@***.***>:﻿
@afkafkafk13 approved this pull request.



In test/Rings/mpoly_affine_algebras.jl:
> ***@***.*** "are_algebraically_independent" begin
+  R, (x, y) = polynomial_ring(QQ, ["x", "y"])
+  V = [x, y]
+  fl, I = are_algebraically_independent(V)
+  @test fl
+  @test is_zero(I)

Please also do an algebraically dependent example in the test.

**Pull request merged**